docs/us.md

```diff
@@ -0,0 +1,352 @@
+
+Then it's turn to issue a block of the first stakeholder as leader of some slot.  
+To be valid for others stakeholders, issued block should be created by rules for last adopted version, so
+updated software of the first stakeholder considers last adopted block version.  
+If last adopted is `0.1.0` then the software creates block using old logic (logic for `0.1.0`)
+otherwise using new logic (logic for `0.2.0`).  
+The last adopted version is `0.1.0`, hence, the first stakeholder issues a block:
+* `BlockVersion` of this block is set to `0.2.0`
+* block created by rules for last adopted block version, i.e. `0.1.0`. 
+
+So the block is issued without new attribute but tagged by the new version.
+Stakeholders with old software expects that all attributes are known because
+last adopted block version equals `lastKnownBlockVersion` for its softwares.
+
+Then the 3rd epoch starts and the threshold (which is regulated by `SoftforkRule`) 
+for adopting new block version is dumped to `0.6`.
+
+Then the second stakeholder downloads this update and when it's turn to issue block, 
+the stakeholder issues block like it has been done by the first stakeholder.
+Also the second one issues one more block as leader, so it produced 2 blocks.
+
+Then 4th epoch starts and sum of issuers stakes is `0.25 + 0.26 = 0.51 < 0.6` 
+(we take into account the second stakeholder only once). Adopting threshold is dumped to the minimal value `0.6`.
+So last adopted block version is still `0.1.0`, `lastKnownBlockVersion` is `0.2.0` for 
+the first and the second stakeholders and `0.1.0` for the third and the fourth.
+
+Then the third stakeholder downloads this update and issues block in the same manner like previous ones.
+
+When 5th epoch starts the sum of issuers stakes is `0.25 + 0.26 + 0.1 = 0.61 > 0.6` and 
+it's time to bump last adopted block version.
+So last adoped version is `0.2.0` now and `lastKnownBlockVersion` is `0.1.0` 
+for the latter stakeholder and `0.2.0` for others.
+
+When it's turn to issue block of the fourth stakeholder, 
+it issues block with `0.1.0` block version and with block format for `0.1.0`.
+Others stakeholders see that last adopted block version equals `0.2.0` and 
+try to decode block using new rules but failed because it doesn't contain attribute with key `228`.
+So this slot left without a block.
+
+You can see despite the provided update system is powerful, it's also very complicated 
+and updates should be handled carefully.
+
+## Verification
+
+### Block header verification
+
+First of all, we check that one of statements for a verifiable block is true:
+* block version equals to last adopted block version. 
+It means the block is created by software where `lastKnownBlockVersion` equals last adopted block version.
+* block version is greater than last adopted block version and this one is confirmed. 
+It means the block is created by software updated by _confirmed_ update proposal.
```

Comment:
I guess it can be simplified to say that block version should be either confirmed or competing. And the code can be simplified as well.

Reply:
how to check it? and why shouldn't we check that it is last adopted?

Reply:
Sorry, I wrote garbage. Either competing or adopted.

Reply:
We have this:
```
-- | Check whether BlockVersion is confirmed.
isConfirmedBV :: MonadPollRead m => BlockVersion -> m Bool
isConfirmedBV = fmap (maybe False bvsIsConfirmed) . getBVState
```

We should rename all usages of `confirmed` to `competing` when we talk about block versions.
Then we have this:
```
-- | Check whether block with given 'BlockVersion' can be created
-- according to current Poll.
--
-- Specifically, one of the following conditions must be true.
-- • Given block version is equal to last adopted block version.
-- • '(major, minor)' of given block version is greater than
-- '(major, minor)' of adopted block version and this block version
-- is confirmed.
canCreateBlockBV :: MonadPollRead m => BlockVersion -> m Bool
canCreateBlockBV bv = do
    lastAdopted <- getAdoptedBV
    isConfirmed <- isConfirmedBV bv
    let toMajMin BlockVersion {..} = (bvMajor, bvMinor)
    return
        (bv == lastAdopted ||
         (toMajMin bv > toMajMin lastAdopted && isConfirmed))
```
The check `toMajMin bv > toMajMin lastAdopted` is likely useless, because if the version is competing, its `(Maj, Min)` must be greater than the last adopted one. Do you agree?

Reply:
I agree, will fix now.

Reply:
done